Mathlib/FieldTheory/Adjoin.lean

```diff
@@ -1388,6 +1388,79 @@ end Induction
 
 end IntermediateField
 
+section Minpoly
+
+open AlgEquiv
+
+variable {K L : Type _} [Field K] [Field L] [Algebra K L]
+namespace AdjoinRoot
+
+/-- The canonical algebraic equivalence between `AdjoinRoot p` and `AdjoinRoot q`, where
+  the two polynomial `p q : K[X]` are equal.-/
+noncomputable def id_algEquiv {p q : K[X]} (hp : p ≠ 0) (hq : q ≠ 0) (h_eq : p = q) :
+    AdjoinRoot p ≃ₐ[K] AdjoinRoot q :=
```

Comment:
Maybe you only need `hp` and `h_eq`? Since `hq` can be derived by them. Same as the following results.

Reply:
You can use `(h_eq ▸ hp)` at the places of `hq` (untested), in this way you don't need that `have`.

Reply:
Thanks!

Reply:
Can you split this construction assuming first of all that one polynomial divides the other (of course it won't be an iso)?

Reply:
> Can you split this construction assuming first of all that one polynomial divides the other (of course it won't be an iso)?

And it can be called `algHomOfDvd` I think.

Reply:
Anyway this file needs to be split... (in another PR)

Reply:
Done!

Reply:
Maybe it has to be split now, the linter is complaining...

Reply:
You can always add a nolint and take care of this later. Or open a PR that does just the splitting.

Reply:
OK, I have added a style exception for now.